shared/js/project.js

```diff
@@ -152,12 +152,21 @@ module.exports = {
   /**
    * Saves the project to the Channels API. Calls `callback` on success if a
    * callback function was provided.
+   * @param {string?} source Optional source to be provided, saving us another
+   *   call to getEditorSource
+   * @param {function} callback Fucntion to be called after saving
    */
   save: function(source, callback) {
+    if (arguments.length === 1) {
```

Comment:
I prefer to still have callback last. In part because if I have code like this
```
dashboard.project.save(function () {
  // some stuff here
}, source);
```
It's pretty easy to not notice that second argument. Also, I think having the callback last is generally the convention in JS.